src/systems/gimbal_controller/GimbalControllerPlugin.cc

```diff
@@ -0,0 +1,484 @@
+                                                 "yawJointName");
+
+	if(_sdf->HasElement("joint_yaw"))
+	{
+		// Add names to map
+		yawJointName = _sdf->Get<std::string>("joint_yaw");
+
+		if(this->dataPtr->model.JointByName(_ecm, yawJointName))
+		{
+			this->dataPtr->yawJoint = this->dataPtr->model.JointByName(_ecm, 
+                                                  "yawJointName");
+		}
+		else
+		{
+			ignwarn << "joint_yaw ["<< yawJointName <<"] does not exist?\n"<< std::endl;
+		}
+	}
+
+	if(!this->dataPtr->yawJoint)
+	{
+		ignerr << "GimbalControllerPlugin::Configure ERROR! Can't get yaw joint." 
+           <<yawJointName <<"' " <<std::endl;
+	}
+
+	// for roll joint
+	std::string rollJointName;
+	this->dataPtr->rollJoint = this->dataPtr->model.JointByName(_ecm, 
+                                             "rollJointName");
+
+	if(_sdf->HasElement("joint_roll"))
+	{
+		// Add names to map
+		rollJointName = _sdf->Get<std::string>("joint_roll");
+
+		if(this->dataPtr->model.JointByName(_ecm, rollJointName))
+		{
+			this->dataPtr->rollJoint = 
+      this->dataPtr->model.JointByName(_ecm, "rollJointName");
+		}
+
+		else
+		{
+			ignwarn << "joint_roll ["<< rollJointName <<"] does not exist?\n"<< std::endl;
+		}
+	}
+
+	if(!this->dataPtr->rollJoint)
+	{
+		ignerr << "GimbalControllerPlugin::Configure ERROR! Can't get roll joint." 
+           <<rollJointName <<"' " <<std::endl;
+	}
+
+	// for pitch joint
+	std::string pitchJointName;
+	this->dataPtr->pitchJoint = 
+  this->dataPtr->model.JointByName(_ecm, "pitchJointName");
+
+	if(_sdf->HasElement("joint_pitch"))
+	{
+		// Add names to map
+		pitchJointName = _sdf->Get<std::string>("joint_pitch");
+		if(this->dataPtr->model.JointByName(_ecm, pitchJointName))
+		{
+			this->dataPtr->pitchJoint = 
+      this->dataPtr->model.JointByName(_ecm, "picthJointName");
+		}
+
+		else
+		{
+			ignwarn << "joint_pitch ["<< pitchJointName <<"] does not exist?\n"<< std::endl;
+		}
+	}
+
+	if(!this->dataPtr->pitchJoint)
+	{
+		ignerr << "GimbalControllerPlugin::Configure ERROR! Can't get pitch joint." 
+           <<pitchJointName <<"' " <<std::endl;
+	}
+
+	// get imu sensor
+	this->dataPtr->imuName = _sdf->Get("imuName", 
+    static_cast<std::string>("imu_sensor")).first;
+}
+
+
+void GimbalControllerPluginPrivate::Init(const EntityComponentManager &_ecm)
+{    
+  // receive pitch command via ignition transport 
+	std::string pitchTopic = this->model.Name(_ecm) + "/gimbal_pitch_cmd";
+  this->node.Subscribe(pitchTopic, 
+    &GimbalControllerPluginPrivate::OnPitchStringMsg);
+
+  // receive roll command via gz transport
+  std::string rollTopic = this->model.Name(_ecm) + "/gimbal_roll_cmd";
+  this->node.Subscribe(rollTopic, 
+    &GimbalControllerPluginPrivate::OnRollStringMsg);
+
+  // receive yaw command via gz transport
+  std::string yawTopic = this->model.Name(_ecm) + "/gimbal_yaw_cmd";
+  this->node.Subscribe(yawTopic, 
+    &GimbalControllerPluginPrivate::OnYawStringMsg);
```

Comment:
Instead of converting string to doubles, we should let users publish the doubles directly. How about offering a single topic that takes an `ignition::msgs::Vector3d` message?